earthquakes.py

```diff
@@ -20,42 +17,80 @@ def get_data():
             "endtime": "2018-10-11",
             "orderby": "time-asc"}
     )
-
-    # The response we get back is an object with several fields.
-    # The actual contents we care about are in its text field:
     text = response.text
-    # To understand the structure of this text, you may want to save it
-    # to a file and open it in VS Code or a browser.
-    # See the README file for more information.
-    ...
-
-    # We need to interpret the text to get values that we can work with.
-    # What format is the text in? How can we load the values?
-    return ...
+    data = json.loads(text)
+    return data["features"]
 
 def count_earthquakes(data):
     """Get the total number of earthquakes in the response."""
-    return ...
+    return len(data)
 
 
 def get_magnitude(earthquake):
     """Retrive the magnitude of an earthquake item."""
-    return ...
+    return earthquake["properties"]["mag"]
 
 
 def get_location(earthquake):
     """Retrieve the latitude and longitude of an earthquake item."""
-    # There are three coordinates, but we don't care about the third (altitude)
-    return ...
+    return earthquake["geometry"]["coordinates"][0:2]
 
 
 def get_maximum(data):
     """Get the magnitude and location of the strongest earthquake in the data."""
-    ...
+    mags = [get_magnitude(quake) for quake in data]
+    max_value = max(mags)
+    index = mags.index(max_value)
+    return [max_value, get_location(data[index])]
 
+def get_year(earthquake):
+    """Gets the year from the given time format. 
+    Uses attributes from the datetime."""
+    timestamp = earthquake["properties"]["time"]
+    return date.fromtimestamp(timestamp/1000).year
 
-# With all the above functions defined, we can now call them and get the result
 data = get_data()
-print(f"Loaded {count_earthquakes(data)}")
-max_magnitude, max_location = get_maximum(data)
-print(f"The strongest earthquake was at {max_location} with magnitude {max_magnitude}")
\ No newline at end of file
+
+# Defines arrays of the years of the earthquakes and empty arrays for the y plots.
+years = np.arange(get_year(data[0]),get_year(data[-1]))
+frequency = np.zeros(len(years))
+av_mags = np.zeros(len(years))
+
+#Loop counting the number of earthquakes per year
+for earthquake in data:
+    for indx, year in enumerate(years):
+        if get_year(earthquake) == year:
+            frequency[indx] +=1
```

Comment:
This could be simplified if you used a dictionary to map years to counts directly. That way you wouldn't need the inner loop of searching for the right year. Similarly for the loops below.

In general, nested loops like this can decrease performance because of the number of iterations they lead to, so try to avoid them when possible.